### PR TITLE
feat: make notification list item injectable

### PR DESCRIPTION
**Pull request by fc-santos**
# Summary of Changes

This PR makes the notification list item injectable so it can be completely customized by other wallets by overriding the default.

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: packages/legacy/core/App/components/listItems/NotificationListItem.tsx [hunk 1]
```diff
@@ -82,7 +82,11 @@ const NotificationListItem: React.FC<NotificationListItemProps> = ({
   const [declineModalVisible, setDeclineModalVisible] = useState(false)
   const [action, setAction] = useState<any>()
   const [closeAction, setCloseAction] = useState<any>()
-  const connection = useConnectionById(notification.connectionId ?? '')
+  const isNotCustomNotificationRecord =
+    notification instanceof BasicMessageRecord ||
+    notification instanceof CredentialExchangeRecord ||
+    notification instanceof ProofExchangeRecord
+  const connection = isNotCustomNotificationRecord ? useConnectionById(notification.connectionId ?? '') : undefined
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/components/listItems/NotificationListItem.tsx]: Hooks should not be called conditionally, it's against the "Rules of Hooks"

**Review by bryce-mcmath — CHANGES_REQUESTED** [review]
Looks ok, just need to fix the conditional hook

File: packages/legacy/core/App/components/listItems/NotificationListItem.tsx (hunk 1 above)
Comment (fc-santos) [reply packages/legacy/core/App/components/listItems/NotificationListItem.tsx]: @bryce-mcmath done!

**Comment by wadeking98** [comment]
I made a quick PR to export the NotificationsListItem component, once that's merged I'll approve this one: https://github.com/MCN-ING/aries-mobile-agent-react-native/pull/16

File: packages/legacy/core/App/components/listItems/NotificationListItem.tsx (hunk 1 above)
Comment (MosCD3) [reply packages/legacy/core/App/components/listItems/NotificationListItem.tsx]: did you resolve this
@fc-santos I can still see the hook conditionally called

Comment (fc-santos) [reply packages/legacy/core/App/components/listItems/NotificationListItem.tsx]: Yes, @MosCD3

**Review by wadeking98 — DISMISSED** [review]

**Review by bryce-mcmath — DISMISSED** [review]

**Review by MosCD3 — DISMISSED** [review]

File: packages/legacy/core/App/screens/Home.tsx [hunk 2]
```diff
@@ -29,6 +29,7 @@ const Home: React.FC<HomeProps> = () => {
     TOKENS.COMPONENT_HOME_FOOTER,
     TOKENS.CONFIG,
     TOKENS.NOTIFICATIONS,
+    TOKENS.NOTIFICATIONS_LIST_ITEM,
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/screens/Home.tsx:33]: I see the token but I don't see the associated item destructured properly, it appears to still be imported at the top of the file as before.

**Review by bryce-mcmath — CHANGES_REQUESTED** [review]

File: packages/legacy/core/App/screens/Home.tsx (hunk 2 above)
Comment (fc-santos) [reply packages/legacy/core/App/screens/Home.tsx:33]: @bryce-mcmath done!

**Review by bryce-mcmath — APPROVED** [review]

**Pull request merged**